$DragonFly: src/gnu/usr.bin/cvs/cvs/Attic/main.c.patch,v 1.2 2005/03/13 11:43:07 corecode Exp $
--- /usr/src/contrib/cvs-1.12.11/src/main.c	2005-01-05 22:17:44.000000000 +0100
+++ main.c	2005-03-07 09:06:36.000000000 +0100
@@ -241,6 +241,7 @@
     "    -q           Cause CVS to be somewhat quiet.\n",
     "    -r           Make checked-out files read-only.\n",
     "    -w           Make checked-out files read-write (default).\n",
+    "    -g           Force group-write perms on checked-out files.\n",
     "    -n           Do not execute anything that will change the disk.\n",
     "    -t           Show trace of program execution (repeat for more\n",
     "                 verbosity) -- try with -n.\n",
@@ -426,7 +427,7 @@
     int help = 0;		/* Has the user asked for help?  This
 				   lets us support the `cvs -H cmd'
 				   convention to give help for cmd. */
-    static const char short_options[] = "+QqrwtnRvb:T:e:d:Hfz:s:xa";
+    static const char short_options[] = "+QqgrwtnRvb:T:e:d:Hfz:s:xa";
     static struct option long_options[] =
     {
         {"help", 0, NULL, 'H'},
@@ -559,6 +560,13 @@
 	    case 'w':
 		cvswrite = 1;
 		break;
+	    case 'g':
+		/*
+		 * Force full group write perms (used for shared checked-out
+		 * source trees, see manual page)
+		 */
+		umask(umask(S_IRWXG|S_IRWXO) & S_IRWXO);
+		break;
 	    case 't':
 		trace++;
 		break;
@@ -662,12 +670,15 @@
     if (argc < 1)
 	usage (usg);
 
+/* This is just too much output */
+/*
     if (readonlyfs && !really_quiet) {
 	error (0, 0,
 	       "WARNING: Read-only repository access mode selected via `cvs -R'.\n\
 Using this option to access a repository which some users write to may\n\
 cause intermittent sandbox corruption.");
     }
+*/
 
     /* Look up the command name. */
 
